### ISSUE TITLE
Processing of comments in XML is buggy

### ISSUE THREAD

**Issue by Kopfbremse** [mikefarah/yq#1446]
**Version of yq**: 4.30.2
**Operating system**: linux
**Installed via**: binary release

**Bug description**
Comments in XML are parsed with type == "!!null".
Order of comments and nodes are not stable.

**Input**
```xml
<root>
  <!-- comment 1 -->
</root>
```
--------------

Undefined type for comments

```bash
yq --input-format xml '.. | type' input.xml
```

**Output**
```text
!!map
!!null
```

--------------

The order of comment and root node are swapped

```bash
yq --input-format xml '.' input.xml
```

**Output**
```text
# comment 1
root:
```

--------------

It is not possible to select the comments for further processing

```bash
yq --input-format xml '.. | select(type == "!!null")' input.xml
```

The command produces no output.

--------------

The comment is selected together with the map

```bash
yq --input-format xml '.. | select(type == "!!map")' input.xml
```

**Output**
```text
# comment 1
root:
```

**Comment by mikefarah** [comment]
Yeah it's been real hard to put the comments in the right place, I'd be surprised if there weren't more issues. 
Will fix in next release

**Comment by mikefarah** [comment]
For the record, comments placed above the map key like that are on the map _key_ node, not the value. You'll need to use triple dots `...` or the `key` operator to select them

**Comment by mikefarah** [comment]
Fixed in 4.30.5
